### PR TITLE
feat: update `eslint-plugin-vue` + update configs

### PR DESCRIPTION
**Pull request by MichaelDeBoey**
Re-submission of https://github.com/mysticatea/eslint-plugin/pull/43

---

BREAKING CHANGE: `es5`, `es2015`, `es2016`, `es2017`, `es2018`, `es2019`, `es2020`, `es2021` & `es2022` configs now have extra rules from `eslint-plugin-vue`

### CODE REVIEW

**Comment by MichaelDeBoey** [comment]
@eslint-community/eslint-tsc @eslint-community/core-team @eslint-community/mysticatea-eslint-plugin Since [this plugin is adhering the same semantic versioning policy as ESLint](https://github.com/eslint-community/eslint-plugin-mysticatea#-semantic-versioning-policy), adding new rules to a plugin would be a breaking change, so this PR would need to be released in a major release I guess?

File: package.json [hunk 1]
```diff
@@ -19,7 +19,7 @@
     "eslint-plugin-eslint-plugin": "^2.3.0",
     "eslint-plugin-node": "^10.0.0",
     "eslint-plugin-prettier": "^3.4.1",
-    "eslint-plugin-vue": "^6.2.2",
+    "eslint-plugin-vue": "^7.20.0",
```
Comment (ota-meshi) [inline package.json]: I would like to use v9 if possible.
If we use v9, use vue-eslint-parser v9 as well.

**Review by ota-meshi — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (MichaelDeBoey) [reply package.json]: v9 would be a breaking change regarding supported Node version (as it drops Node 12) and I think we should keep the supported Node versions with what ESLint itself supports (even though Node 12 is now EOL).

I was planning on upgrading to `eslint-plugin-vue` v8 together with `vue-eslint-parser` in a follow-up PR

Comment (ota-meshi) [reply package.json]: You are right. I didn't check the node version 😓

**Review by ota-meshi — APPROVED** [review]

**Pull request merged**